src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -572,7 +572,12 @@ reconnectSMPClient n tc c tSess@(_, srv, _) = do
         unless (null conns) $ notifySub "" $ UP srv conns
       let (tempErrs, finalErrs) = partition (temporaryAgentError . snd) errs
       liftIO $ mapM_ (\(connId, e) -> notifySub connId $ ERR e) finalErrs
-      mapM_ (throwError . snd) $ listToMaybe tempErrs
+      forM_ (listToMaybe tempErrs) $ \(_, err) ->
+        if allowClose && null okConns && S.null cs && null finalErrs
+          then do
+            liftIO $ closeClient c smpClients tSess
+            throwError $ BROKER (B.unpack $ strEncode srv) (RESPONSE "TODO: new error for no progress")
+          else throwError err
```

Comment:
```suggestion
      forM_ (listToMaybe tempErrs) $ \(_, err) ->
        liftIO . when (allowClose && null okConns && S.null cs && null finalErrs) $
          closeClient c smpClients tSess
        throwError err
```

Reply:
I think we can just throw temp error as usual, as the log will show disconnect/connect event in this case, so we know it's working - no need for new error.